### DISCUSSION TITLE
Bug on nativeUpdate

### DISCUSSION THREAD

**Discussion by DavideCarvalho** [mikro-orm/mikro-orm#4337] [Q&A]
**Describe the bug**

Trying to update multiples rows with nativeUpdate, but the query seems broken.

It looks like id tries to change the id of the row as well, even though I'm not doing that

**Stack trace**
```
"update \"approval_flow\" set \"device_application_id\" = '{\"status\":\"PENDING_REVIEW\",\"updatedBy\":\"c53e44c6-93e7-4609-91d0-f1d8d979facd\"}' where \"id\" in (select \"a0\".\"id\" from (select distinct \"a0\".\"id\" from \"approval_flow\" as \"a0\" left join \"device_transmission_revision\" as \"d1\" on \"a0\".\"device_application_id\" = \"d1\".\"id\" left join \"device_transmission\" as \"d2\" on \"d1\".\"device_transmission_id\" = \"d2\".\"id\" left join \"document\" as \"d3\" on \"d2\".\"id\" = \"d3\".\"device_transmission_id\" where \"d2\".\"id\" = '073286ef-8c18-413a-9ce3-d3e2b5ef6f69' and \"d3\".\"id\" is not null) as \"a0\") - invalid input syntax for type uuid: \"{\"status\":\"PENDING_REVIEW\",\"updatedBy\":\"c53e44c6-93e7-4609-91d0-f1d8d979facd\"}\""
```

**To Reproduce**
1. ...
2. ...
3. ...

**Expected behavior**

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 19.6.0 |
| typescript | 4.9.3 |
| mikro-orm | 5.5.3 |
| @mikro-orm/postgresql | 5.5.3 |

Query I'm doing

![image](https://github.com/mikro-orm/mikro-orm/assets/15630886/07e98940-1a56-45fc-bbb5-21b1ac578279)

Error

![image](https://github.com/mikro-orm/mikro-orm/assets/15630886/f4f7ee0b-ff71-436d-9538-f30ef4920e2a)

**Comment by B4nan** [comment]
How does your entity definition look like? 

And be sure to update to latest version first, you are quite behind with v5.5.3

**Comment by DavideCarvalho** [comment]
```typescript
@Entity({
  tableName: 'approval_flow',
  customRepository: () => ApprovalFlowRepository,
})
export class ApprovalFlowEntity extends BaseEntity {
  [EntityRepositoryType]?: ApprovalFlowRepository;

  @ManyToOne({
    entity: () => DeviceTransmissionRevisionEntity,
    nullable: false,
    trackChanges: false,
  })
  deviceApplication: DeviceTransmissionRevisionEntity;
}
```

```typescript
@Entity({
  tableName: 'device_transmission_revision',
  customRepository: () => DeviceTransmissionRevisionRepository,
})
export class DeviceTransmissionRevisionEntity extends BaseEntity {
  [EntityRepositoryType]?: DeviceTransmissionRevisionRepository;

    @Enum({
    items: () => DeviceTransmissionRevisionStatus,
    nullable: true,
    columnType: 'text',
    trackChanges: false,
  })
  status?: DeviceTransmissionRevisionStatus;
}
```

updatedBy is part of the BaseEntity

Also, just updated to "5.7.5", but the issue remains

**Reply by DavideCarvalho** [reply]
I just copied the fields related to the query, if needed, I can copy the full entity definition

**Reply by B4nan** [reply]
Looks like you are trying to set a nested property (a property of a relation - so a value that lives in a different table), you can't do that with `nativeUpdate`, that is basically just a shortcut for query builder, it won't handle nesting.

**Reply by DavideCarvalho** [reply]
Oh, got it, okay. I thought it would update only those values

Maybe this should be in the docs - I don't know if it is, but if don't, it would be helpful to make it explicit
